x/storage/keeper/msg_server.go

```diff
@@ -362,7 +362,13 @@ func (k msgServer) MirrorObject(goCtx context.Context, msg *types.MsgMirrorObjec
 
 	operator := sdk.MustAccAddressFromHex(msg.Operator)
 
-	objectInfo, found := k.Keeper.GetObjectInfoById(ctx, msg.Id)
+	var objectInfo *types.ObjectInfo
+	found := false
+	if msg.Id.GT(sdk.NewUint(0)) {
+		objectInfo, found = k.Keeper.GetObjectInfoById(ctx, msg.Id)
+	} else {
+		objectInfo, found = k.Keeper.GetObjectInfo(ctx, msg.BucketName, msg.ObjectName)
```

Comment:
It's a bit confused. Any checks will added to make sure the ID and name correspond to each other? Or if the name is specified, the id must be zero, Or if the ID is greater than 0, the name must be empty.

Reply:
Yes. The check had been added to the `ValidateBasic` functions.